multimap/src/main/java/org/infinispan/multimap/impl/SortedSetBucket.java

```diff
@@ -120,6 +121,39 @@ public Collection<ScoredValue<V>> inter(Collection<ScoredValue<V>> inputValues,
       return sortedMergeScoredValues;
    }
 
+   public List<ScoredValue<V>> randomMembers(int count) {
+      if (count == 1 || count == -1) {
+         Random random = new Random();
+         int rank = random.nextInt(scoredEntries.size());
+         return this.subsetByIndex(rank, rank, false);
+      }
+
+      if (count < 0) {
+         int totalCount = Math.abs(count);
+         List<ScoredValue<V>> randomEntries = new ArrayList<>(totalCount);
+         Random random = new Random(entries.size());
+         random.ints(totalCount, 0, entries.size())
+               .forEach(e -> randomEntries.add(this.subsetByIndex(e, e, false).get(0)));
+         return randomEntries;
+      }
```

Comment:
I suggest we do this even for positive counters but limit on `Math.min(Math.abs(count), entries.size())`. We do the padding to have the `count` number of elements on the caller's side. It would help a bit with the remote calls.

And a side note, I suggest we replace the Random instantiation and use the ThreadLocalRandom directly.

Reply:
@jabolina + counters don't allow duplicates, and negative counters do allow them

Reply:
I reported the change concerning the thread local random

Reply:
Thanks. I think would be nice to include the changes to limit the return and compute the duplicates on the caller's side.

Reply:
this looks more like an early optimization for me, but I might be completely wrong. do you believe is necessary to address that now?

Reply:
Oh, sure. We can do it later.